sources/api/bootstrap-containers/src/main.rs

```diff
@@ -0,0 +1,708 @@
+            }
+
+            // Subcommands
+            "create-containers" | "mark-bootstrap"
+                if subcommand.is_none() && !arg.starts_with('-') =>
+            {
+                subcommand = Some(arg)
+            }
+
+            // Other arguments are passed to the subcommand parser
+            _ => subcommand_args.push(arg),
+        }
+    }
+
+    match subcommand.as_deref() {
+        // Default subcommand is 'create-containers'
+        None | Some("create-containers") => Ok((global_args, Subcommand::CreateContainers {})),
+        Some("mark-bootstrap") => Ok((global_args, parse_mark_bootstrap_args(subcommand_args)?)),
+        Some(x) => {
+            return error::Usage {
+                message: format!("Unknown subcommand '{}'", x),
+            }
+            .fail()
+        }
+    }
+}
+
+/// Parses arguments for the 'mark-boostrap' subcommand
+fn parse_mark_bootstrap_args(args: Vec<String>) -> Result<Subcommand> {
+    let mut container_id = None;
+    let mut mode = None;
+
+    let mut iter = args.into_iter();
+    while let Some(arg) = iter.next() {
+        match arg.as_ref() {
+            "--container-id" => {
+                container_id = Some(iter.next().context(error::Usage {
+                    message: "Did not give argument to --container-id",
+                })?);
+            }
+
+            "--mode" => {
+                mode = Some(iter.next().context(error::Usage {
+                    message: "Did not give argument to --mode",
+                })?);
+            }
+
+            x => {
+                return error::Usage {
+                    message: format!("Unexpected argument '{}'", x),
+                }
+                .fail()
+            }
+        }
+    }
+
+    let container_id = container_id.context(error::Usage {
+        message: format!("Did not give argument to --container-id"),
+    })?;
+
+    let mode = mode.context(error::Usage {
+        message: format!("Did not give argument to --mode"),
+    })?;
+
+    Ok(Subcommand::MarkBootstrap(MarkBootstrapArgs {
+        container_id: container_id,
+        // Fail if 'mode' is invalid
+        mode: ValidBootstrapContainerMode::try_from(mode.to_string())
+            .context(error::BootstrapContainerMode)?,
+    }))
+}
+
+/// Handles how the bootstrap containers' systemd units are created
+fn handle_bootstrap_container<S>(
+    name: S,
+    container_details: &model::BootstrapContainer,
+) -> Result<()>
+where
+    S: AsRef<str>,
+{
+    let name = name.as_ref();
+
+    info!("Handling bootstrap container '{}'", name);
+
+    // Get basic settings, as retrieved from API.
+    let source = container_details
+        .source
+        .as_ref()
+        .context(error::MissingField {
+            name,
+            field: "source",
+        })?;
+
+    let mode = container_details
+        .mode
+        .as_ref()
+        .context(error::MissingField {
+            name,
+            field: "mode",
+        })?;
```

Comment:
This creates the same concern as #1385 for bootstrap containers.

I would say:
* if no mode, default to "off"
* if not essential, default to "false" (as you've done)

Reply:
I'm still working on fixing this